### ISSUE TITLE
directory-local org-roam-directy value of "." becomes global

### ISSUE THREAD

**Issue by codygman** [org-roam/org-roam#1459]
### Description

If I use the recommended .dir-locals in a secondary org-roam folder:

```
((nil . ((org-roam-directory . ".")
         (org-roam-db-location . "./org-roam.db"))))
```

I somehow end up with `org-roam-directory` being globally set to `.` which causes all kinds of issues like running find calls on my entire $HOME directory.

#### Steps to Reproduce

1. Create new folder with `.dir-locals.el` above and add an org roam file
2. Visit that a file in that folder
3. Go back to your home directory and `C-h v org-roam-directory RET`

You'll likely see something like:

```
Its value is "."
Original value was 
"/home/cody/org-roam/"
Local in buffer 20210326144748-org_roam.org; global value is 
"/home/cody/org-roam/"

```

If not, wait until the update function runs. I think something in there is globally setting this value.

#### Expected Results

My org-roam-directory variable is *never* set to `.` and org-roam-db-update isn't run in every buffer I visit.

#### Actual Results

If I load that directory local variable it effectively means any directory I visit will get an org-roam.db and be updated on idle timer.

### Environment
Copy info below this line into issue:
- Emacs: GNU Emacs 28.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.24, cairo version 1.16.0)
- Framework: N/A
- Org: Org mode version 9.4.4 (9.4.4-27-gb712b9-elpa @ /nix/store/fslr7dhwzq7hi406292wsqb8h1kv4hk6-emacs-packages-deps/share/emacs/site-lisp/elpa/org-20210315/)
- Org-roam: 1.2.3

- Org-roam commit: https://github.com/jethrokuan/org-roam/commit/commithashhere

Not sure how to get the commit hash, but I'm using `org-roam-20210308.457`.

**Comment by codygman** [comment]
I also have some notes on trying to work through the issue and debug it here: https://github.com/codygman/tech-roam/blob/master/20210326144748-org_roam.org#using-dir-locals-with-org-roam-somehow-results-in-org-roam-directory-being-set-globally-to-

**Comment by codygman** [comment]
I think this could be specific to my setup and involve a weird interaction with with-editor, though that's just a theory.

**Comment by codygman** [comment]
Reproduction steps are:

1. Go into org-roam directory, `org-roam-db-build-cache`.
2. create `org-roam-directory` and add .dir-locals.el with these contents
```emacs-lisp
((nil . ((org-roam-directory . ".")
         (org-roam-db-location . "./org-roam.db"))))
```
3. exit/re-enter the directory to trigger it's .dir-locals.el
4. make /tmp/foobar and create /tmp/foobar/something.org
5. open /tmp/foobar/something.org and wait a bit, `/tmp/foobar/org-roam.db` will be generated

**Comment by codygman** [comment]
It looks like others take a different approach such as:

```emacs-lisp
((nil . ((eval . (setq-local org-roam-directory (expand-file-name "./"))))))
```

**Comment by codygman** [comment]
I will try replacing my files with this:

```
((nil . (
	 (eval . (setq-local org-roam-directory (expand-file-name "./")))
	 (eval . (setq-local org-roam-db-location (expand-file-name "./org-roam.db"))))))
```

**Comment by jethrokuan** [comment]
SFLR, I updated the docs I updated the docs to reflect the correct `.dir-locals.el` usage. https://github.com/org-roam/org-roam/commit/997ddcbf4b0373bb449d09f1db6cf5de6983de5e

The TLDR is that `org-roam-directory` and `org-roam-db-location` should be an absolute path, not relative, which is why the `expand-file-name` calls were necessary.

**Comment by jethrokuan** [comment]
Closed with 997ddcb
